carbon/config.go

```diff
@@ -113,6 +113,7 @@ type carbonserverConfig struct {
 	FailOnMaxGlobs    bool      `toml:"fail-on-max-globs"`
 	MetricsAsCounters bool      `toml:"metrics-as-counters"`
 	TrigramIndex      bool      `toml:"trigram-index"`
+	TrieIndex         bool      `toml:"trie-index"`
```

Comment:
Since trie and trigram are mutually exclusive we need to prevent them from being enabled both at the same time.

Reply:
@azhiltsov you are fast as lightning. :D

When trie-index is enabled, trigram-index should be ignored in the code. But I need to double check.